File: Python/LAB2_POO/Direccion.py
class Direccion:
    def __init__(self,calle,nomenclatura,barrio,ciudad,edificio = None,apto =None):
        self.__calle = calle
        self.__nomenclatura = nomenclatura 
        self.__barrio = barrio
        self.__ciudad = ciudad 
        self.__edificio = edificio
        self.__apto = apto 

    def getCalle(self):
        return self.__calle
    
    def setCalle(self,c):
        self.__calle = c

    def getNomenclatura(self):
        return self.__nomenclatura
    
    def setNomenclatura(self,n):
        self.__nomenclatura = n
    
    def getBarrio(self):
        return self.__barrio
    
    def setBarrio(self,b):
        self.__barrio = b
    
    def getCiudad(self):
        return self.__ciudad
    
    def setCiudad(self, ci):
        self.__ciudad = ci
    
    def getEdificio(self):
        return self.__edificio
    
    def setEdificio(self, e):
        self.__edificio = e
    
    def getApto(self):
        return self.__apto
    
    def setApto(self,a):
        self.__apto = a


    def __str__(self) :
        return f'{self.__calle} # {self.__nomenclatura} {self.__barrio}  {self.__ciudad} {self.__edificio or""} apto/complemento {self.__apto or""}'
    

File: Java/LAB2_revision_POO/Lab2/src/lab2/Direccion.java

package lab2;


public class Direccion {
    //Atributos
    private String calle;
    private String nomenclatura;
    private String barrio;
    private String ciudad;
    private String edificio;
    private String apto;
    
    //Constructores
    public Direccion(){
    }
    
    public Direccion(String calle,String nomenclatura,String barrio, String ciudad, String edificio, String apto){
        this.calle = calle;
        this.nomenclatura = nomenclatura;
        this.barrio = barrio;
        this.ciudad = ciudad;
        this.edificio = edificio;
        this.apto = apto;

    }
    
    //Getters and setters

    public String getCalle() {
        return calle;
    }

    public void setCalle(String calle) {
        this.calle = calle;
    }

    public String getNomenclatura() {
        return nomenclatura;
    }

    public void setNomenclatura(String nomenclatura) {
        this.nomenclatura = nomenclatura;
    }
    
    public String getBarrio(){
        return this.barrio;
    }
    
    public void setBarrio(String barrio){
        this.barrio = barrio;
    }

    public String getCiudad() {
        return ciudad;
    }

    public void setCiudad(String ciudad) {
        this.ciudad = ciudad;
    }

    public String getEdificio() {
        return edificio;
    }

    public void setEdificio(String edificio) {
        this.edificio = edificio;
    }

    public String getApto() {
        return apto;
    }

    public void setApto(String apto) {
        this.apto = apto;
    }
    
    //Método toString

    @Override
    public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append(calle);
    sb.append(nomenclatura);
    sb.append(" barrio: ").append(barrio);
    sb.append(", Ciudad: ").append(ciudad);
    
    // Verifica si hay un edificio y un apartamento
    if (edificio != null && !edificio.isEmpty()) {
        sb.append(", edificio : ").append(edificio);
    }
    if (apto != null && !apto.isEmpty()) {
        sb.append(" apto: ").append(apto);
    }
    return sb.toString();
}
    

}
